web/src/unread_ops.ts

```diff
@@ -648,10 +698,52 @@ export function mark_topic_as_unread(stream_id: number, topic: string): void {
     );
 }
 
-export function mark_all_as_read(modal_id?: string): void {
+export function mark_all_as_read(modal_id: string): void {
     bulk_update_read_flags_for_narrow(all_unread_messages_narrow, "add", {}, modal_id);
 }
 
+export function mark_muted_topic_messages_as_read(modal_id: string): void {
+    const unread_topics_count = unread.get_unread_topics().topic_counts;
+    const muted_topic_unread_messages_count = unread.get_counts().muted_topic_unread_messages_count;
+
+    // In case there are no muted topics with unread messages, we can
+    // close the dialog box immediately.
+    if (!muted_topic_unread_messages_count) {
+        modals.close_if_open(modal_id);
+        return;
+    }
+
+    for (const [stream_id, topics] of unread_topics_count) {
+        const stream_name = stream_data.get_stream_name_from_id(stream_id);
+        for (const [topic_name] of topics) {
+            if (is_topic_muted(stream_id, topic_name)) {
+                bulk_update_read_flags_for_narrow(
+                    [
+                        {operator: "stream", operand: stream_name},
+                        {operator: "topic", operand: topic_name},
+                    ],
+                    "add",
+                    {},
+                    modal_id,
+                );
+            }
+        }
```

Comment:
I think we can do a query similar to the one below for muted topic messages. This query is also not accurate in the terms that it relies on locally cached data to mark messages unread which might not be accurate if we have missing unreads data locally.

Reply:
Hey @amanagr, I don’t think it’s possible to construct a combination of narrow options to mark messages in muted topics as read. Please let me know if I’m mistaken. I also have another approach that might meet our requirements.

Could we use get_user_topics_for_visibility_policy to retrieve all muted topics and then use mark_topic_as_read to mark the unread ones as read? This would require an additional loop to iterate through each topic and mark it as read.